### PR TITLE
Make sure to always delete temp dir

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java [hunk 1]
```diff
@@ -626,8 +627,13 @@ public ApplicationPackage preprocess(Zone zone, DeployLogger logger) throws IOEx
             // if it fails (with DirectoryNotEmptyException (!)) we need to use 'copy' instead
             // (this will always be the case for the application package for a standalone container).
             Files.move(tempDir, preprocessedDir.toPath());
+            tempDir = null;
         } catch (AccessDeniedException | DirectoryNotEmptyException e) {
             preprocess(appDir, preprocessedDir, zone);
+            IOUtils.recursiveDeleteDir(tempDir.toFile());
```
Comment (baldersheim) [inline config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java:633]: Should you remove this one and always do the cleanup in the finally clause, if not I guess you should set tempDir to null here too.

Comment (hakonhall) [inline config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java:633]: Actually the deletion will naturally be done in the finally, so this statement can be removed.

**Review by hakonhall — DISMISSED** [review]

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java (hunk 1 above)
Comment (hmusum) [reply config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java:633]: Yes, I didn't intend this one to be there, mixed up when editing, will fix

Comment (hmusum) [reply config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationPackage.java]: See comment below

**Review by hakonhall — APPROVED** [review]

**Review by baldersheim — APPROVED** [review]

**Pull request merged**